docs/csharp/proto2.md

```diff
@@ -0,0 +1,184 @@
+Proto2 support in Google.Protobuf is finally here! This document outlines the new changes brought about 
+by this initial release of proto2 support. The generated code and public API associated with proto2 
+is experimental and subject to change in the future. APIs may be added, removed, or adjusted as feedback is received.
+Generated code may also be modified, adding, removing, or adjusting APIs as feedback is received.
+
+# Generated code
```

Comment:
it might be useful to add a small section "Enabling proto2 features"  that says that for proto3 users, nothing is changing and that to use proto2, one needs to have .proto files that specify syntax = proto2.
Also mention that proto3 is the recommended version of protobuf and everyone should use that unless they have special needs (perhaps link to proto3 docs). Proto2 support is there for more advanced use cases (e.g. working with proto descriptors that are historically proto2) or to interop with legacy systems that use proto2.

Reply:
Also link to proto2 docs?